osmnx/pois.py

```diff
@@ -37,36 +38,116 @@ def parse_poi_query(north, south, east, west, amenities=None, timeout=180, maxsi
+        List of amenities that will be used for finding the POIs from the
+        selected area. Results returned are the _union_, not _intersection_ of
+        each individual amenity tag; each result matches at least one tag given.
+        Equivalent to `tags={'amenity': amenities}`. Some POIs don't use the
+        `amenity` key, in which case you should use the `tags` option
+    tags : dict[str, Union[bool, str, List[str]]]
+        Dict of tags used for finding POIs from the selected area. Results
+        returned are the _union_, not _intersection_ of each individual tag;
+        each result matches at least one tag given.
+
+        The keys should be the OSM tag key, (i.e. `amenity`, `landuse`,
+        `highway`, etc) and the values should be `True` to retrieve all objects
+        with the given tag, a string to get a single tag-value combination, and
+        a list of strings to get multiple values for the given tag. For example,
+
+            tags = {
+                'amenity':True,
+                'landuse':['retail','commercial'],
+                'highway':'bus_stop'
+            }
+
+        Would return all amenities, `landuse=retail` and `landuse=commercial`,
+        and `highway=bus_stop`. If both `amenities` and `tags` are combined, any
+        `amenity` tags will be combined.
     timeout : int
         Timeout for the API request.
     """
-    if amenities:
-        # Overpass QL template
-        query_template = ('[out:json][timeout:{timeout}]{maxsize};((node["amenity"~"{amenities}"]({south:.6f},'
-                          '{west:.6f},{north:.6f},{east:.6f});(._;>;););(way["amenity"~"{amenities}"]({south:.6f},'
-                          '{west:.6f},{north:.6f},{east:.6f});(._;>;););(relation["amenity"~"{amenities}"]'
-                          '({south:.6f},{west:.6f},{north:.6f},{east:.6f});(._;>;);););out;')
-
-        # Parse amenties
-        query_str = query_template.format(amenities="|".join(amenities), north=north, south=south, east=east, west=west,
-                                          timeout=timeout, maxsize=maxsize)
-    else:
-        # Overpass QL template
-        query_template = ('[out:json][timeout:{timeout}]{maxsize};((node["amenity"]({south:.6f},'
-                          '{west:.6f},{north:.6f},{east:.6f});(._;>;););(way["amenity"]({south:.6f},'
-                          '{west:.6f},{north:.6f},{east:.6f});(._;>;););(relation["amenity"]'
-                          '({south:.6f},{west:.6f},{north:.6f},{east:.6f});(._;>;);););out;')
-
-        # Parse amenties
-        query_str = query_template.format(north=north, south=south, east=east, west=west,
-                                          timeout=timeout, maxsize=maxsize)
-
-    return query_str
-
-
-def osm_poi_download(polygon=None, amenities=None, north=None, south=None, east=None, west=None,
-                     timeout=180, max_query_area_size=50*1000*50*1000):
+    # Convert input into dict of tags
```

Comment:
Lines 68-98 could use some more comments. First at the top it'd be useful to document more clearly what this chunk of code does and why. Second, throughout this chunk of code it'd be useful to comment more clearly what the line's logic is. It's a little opaque right now.

Reply:
I think I added a lot of comments, and simplified the code a bit as well, at the risk of being too verbose.